tools/integration_tests/readonly/stat_object_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for file operations with --o=ro flag set.
+package readonly_test
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/setup"
+)
+
+func statExistingObj(objPath string, t *testing.T) (file os.FileInfo) {
+	file, err := os.Stat(objPath)
+	if err != nil {
+		t.Errorf("Fail to stat the object.")
+	}
+	return file
+}
+
+func TestStatFile(t *testing.T) {
+	filePath := path.Join(setup.MntDir(), FileNameInTestBucket)
+	file := statExistingObj(filePath, t)
+
+	if file.Name() != FileNameInTestBucket || file.IsDir() != false {
+		t.Errorf("Stat incorrrect file.")
+	}
+}
+
+func TestStatFileFromSubDirectory(t *testing.T) {
+	filePath := path.Join(setup.MntDir(), DirectoryNameInTestBucket, FileInSubDirectoryNameInTestBucket)
+	file := statExistingObj(filePath, t)
+
+	if file.Name() != FileInSubDirectoryNameInTestBucket || file.IsDir() != false {
+		t.Errorf("Stat incorrrect file.")
+	}
+}
+
+func TestStatDirectory(t *testing.T) {
+	DirPath := path.Join(setup.MntDir(), DirectoryNameInTestBucket)
+	dir := statExistingObj(DirPath, t)
+
+	if dir.Name() != DirectoryNameInTestBucket || dir.IsDir() != true {
+		t.Errorf("Stat incorrrect Directory.")
+	}
+}
+
+func TestStatSubDirectory(t *testing.T) {
+	DirPath := path.Join(setup.MntDir(), DirectoryNameInTestBucket, SubDirectoryNameInTestBucket)
+	dir := statExistingObj(DirPath, t)
+
+	if dir.Name() != SubDirectoryNameInTestBucket || dir.IsDir() != true {
+		t.Errorf("Stat incorrrect Directory.")
+	}
+}
+
+func statNotExistingObj(objPath string, t *testing.T) {
```

Comment:
Please change the name of this function according to the comment on this PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1069/files#r1175119688